code/mob.dm

```diff
@@ -1538,6 +1540,54 @@
 		src.health = max_health
 		setalive(src)
 
+/// Adds a 16-length color matrix to the mob's list of color matrices
+/// cmatrix is the color matrix (must be a 16-length list!), label is the string to be used for dupe checks and removal
+/mob/proc/apply_color_matrix(var/list/cmatrix, var/label)
```

Comment:
It's a bit confusing that this proc seems to sorta act as if the order of the matrices didn't matter when it actually can matter a lot. However, this is unlikely to be a problem in practice hopefully.

Reply:
Its partly because I haven't thought up a good way to handle applying certain matrices in a specific order. And yeah, the order the matrices are multiplied is important, but right now there doesn't seem to be anything that'd rely on this too heavily, where a specific expected result is required. I imagine if someone wanted to do something like have an adventure puzzle that needs multiple pieces of vision-altering gear produce a specific color matrix to solve, it might make that kinda tricky. I could try figuring something out for that when the need arises, though!

And its also so that stacking different colorblindness effects makes a different result depending on the order applied, so people can experiment with different combinations.